api/src/server-utils.js

```diff
@@ -19,7 +19,7 @@ var writeJSON = function(res, code, message, details) {
 };
 
 var respond = function(req, res, code, message, details) {
-  if (wantsJSON(req)) {
+  if (wantsJSON(req) || typeof message === 'object') {
```

Comment:
I am happy with this, though because it might have further reaching implications I'd like to get a second opinion. Tagging in @garethbowen: are you OK with us returning JSON when someone hasn't asked for it explicitly? Are we being bad netizens if we do that?

Reply:
It's either this or stringify-ing the message ourselves. 
Would your netizen conscience be clear then? :)

Reply:
I agree with Stefan - I worry about side-effects here.

As far as I can tell the two services in the EditUser controller that could end up here are both passing `'Accept': 'application/json',` which should pass the `wantsJSON` test anyway, so it should just work?

Reply:
Ok sure. Medic-conf also creates users, and it also sends an `accept` header.
Our E2E user creation don't send an `accept` header, that's where I saw this error. 

This is what happens. 
```shell
ERROR: UNHANDLED REJECTION! 
ERROR:   Reason: { TypeError [ERR_INVALID_ARG_TYPE]: The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type object
    at Function.from (buffer.js:232:9)
    at ServerResponse.end (/home/diana/medic/medic-webapp/api/node_modules/compression/index.js:115:29)
    at respond (/home/diana/medic/medic-webapp/api/src/server-utils.js:33:7)
    at Object.error (/home/diana/medic/medic-webapp/api/src/server-utils.js:67:5)
    at auth.check.then.then.catch.err (/home/diana/medic/medic-webapp/api/src/controllers/users.js:58:33)
    at process._tickCallback (internal/process/next_tick.js:68:7)
  [stack]:
   'TypeError [ERR_INVALID_ARG_TYPE]: The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type object\n    at Function.from (buffer.js:232:9)\n    at ServerResponse.end (/home/diana/medic/medic-webapp/api/node_modules/compression/index.js:115:29)\n    at respond (/home/diana/medic/medic-webapp/api/src/server-utils.js:33:7)\n    at Object.error (/home/diana/medic/medic-webapp/api/src/server-utils.js:67:5)\n    at auth.check.then.then.catch.err (/home/diana/medic/medic-webapp/api/src/controllers/users.js:58:33)\n    at process._tickCallback (internal/process/next_tick.js:68:7)',
  [message]:
   'The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type object' } 
```

and the client receives an empty response after timeout.
I could just update user-creation in our e2e tests.

Reply:
Sounds good.

Reply:
Great. I'll close this and include this change in e2e along with the sentinel stuff which is in the making.